tt_metal/impl/event/event.hpp

```diff
@@ -13,5 +16,19 @@ namespace tt::tt_metal
         Device * device = nullptr;
         uint32_t cq_id = -1;
         uint32_t event_id = -1;
+        bool ready = false; // Event is ready for use.
+
+        // With async CQ, must wait until event is populated by child thread before using.
+        // Opened #5988 to track removing this, and finding different solution.
+        void wait_until_ready() {
+            while (!ready) {
```

Comment:
This doesn't seem thread safe, I would wrap `bool ready` inside `std::atomic<bool> ready` then you should be good to go

Reply:
Whoops, thanks for catching. Updated and squashed that change.